File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        big = 0
        secBig = 0
        for num in nums:
            if num > big:
                secBig = big
                big = num
            else:
                secBig = max(secBig, num)
        return (big - 1) * (secBig - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int big = 0;
        int secBig = 0;
        for (int num: nums) {
            if (num > big) {
                secBig = big;
                big = num;
            }
            else {
                secBig = max(secBig, num);
            }
        }
        return (big - 1) * (secBig - 1);
    }
};